app/(frontend)/(app)/tweets/[id]/page.tsx

```diff
@@ -0,0 +1,68 @@
+'use client'
+
+import {useParams, useRouter} from "next/navigation";
+import {useCallback, useEffect, useState} from "react";
+import useNotification from "@/app/ui/hooks/useNotification";
+import {NotificationType} from "@/app/ui/context/NotificationContext";
+import {motion} from "framer-motion";
+import useQuery, {IQueryParams} from "@/app/ui/hooks/useQuery";
+import {ITweet} from "@/app/lib/definitions";
+import Tweet from "@/app/ui/Tweet";
+
+export default function TweetPage() {
+
+    const { id } = useParams<{ id: string }>()
+    const { setNotification } = useNotification()
+    const router = useRouter();
+    const query$ = useQuery();
+
+    const [
+        tweet,
+        setTweet
+    ] = useState<ITweet>()
+
+    useEffect(() => {
```

Comment:
**it could be put into its own hook**
Also, put a `console.log` in this `useEffect` and check how many **re-renders** does it have:
- it is typical that when we have a **dependency array** with more than 2 element it has **too many re-renders in the background** which we don't notice at first, but it does **impact the performance**
- if it happens, then you have to debug which **dependency or dependencies** triggers these **re-renders**
- when you find these **dependencies**, put them in a `useCallback` or `useMemo`
- and/or you can have multiple `useEffect`s to separate the logics